samples/python/54.teams-task-module/bots/teams_task_module_bot.py

```diff
@@ -2,84 +2,162 @@
+            # on_teams_task_module_submit.
+            task_info.card = TeamsTaskModuleBot.__create_adaptive_card_attachment()
+            TeamsTaskModuleBot.__set_task_info(
+                task_info, TaskModuleUIConstants.ADAPTIVE_CARD
             )
-        )
-        card = CardFactory.adaptive_card(
-            {
-                "version": "1.0.0",
-                "type": "AdaptiveCard",
-                "body": [
-                    {"type": "TextBlock", "text": "Enter Text Here",},
-                    {
-                        "type": "Input.Text",
-                        "id": "usertext",
-                        "placeholder": "add some text and submit",
-                        "IsMultiline": "true",
-                    },
-                ],
-                "actions": [{"type": "Action.Submit", "title": "Submit",}],
-            }
-        )
 
-        task_info = TaskModuleTaskInfo(
-            card=card, title="Adaptive Card: Inputs", height=200, width=400
-        )
-        continue_response = TaskModuleContinueResponse(type="continue", value=task_info)
-        return TaskModuleResponse(task=continue_response)
+        return TaskModuleResponseFactory.to_task_module_response(task_info)
 
     async def on_teams_task_module_submit(
         self, turn_context: TurnContext, task_module_request: TaskModuleRequest
     ) -> TaskModuleResponse:
+        """
+        Called when data is being returned from the selected option (see `on_teams_task_module_fetch').
+        """
+
+        # Echo the users input back.  In a production bot, this is where you'd add behavior in
+        # response to the input.
         await turn_context.send_activity(
             MessageFactory.text(
                 f"on_teams_task_module_submit: {json.dumps(task_module_request.data)}"
             )
         )
 
-        message_response = TaskModuleMessageResponse(type="message", value="Thanks!")
+        message_response = TaskModuleMessageResponse(value="Thanks!")
         return TaskModuleResponse(task=message_response)
 
-    def _get_task_module_hero_card(self) -> Attachment:
-        task_module_action = CardAction(
-            type="invoke",
-            title="Adaptive Card",
-            value={"type": "task/fetch", "data": "adaptivecard"},
-        )
-        card = HeroCard(
-            title="Task Module Invocation from Hero Card",
-            subtitle="This is a hero card with a Task Module Action button.  Click the button to show an Adaptive "
-                     "Card within a Task Module.",
-            buttons=[task_module_action],
-        )
+    @staticmethod
+    def __set_task_info(task_info: TaskModuleTaskInfo, ui_constants: UISettings):
+        task_info.height = ui_constants.height
+        task_info.width = ui_constants.width
+        task_info.title = ui_constants.title
+
+    @staticmethod
+    def __get_task_module_hero_card_options() -> Attachment:
+        buttons = [
+            CardAction(
```

Comment:
When I ran this, I had to json.dumps the value:

```python
CardAction(
                type="invoke",
                title=card_type.button_title,
                value=json.dumps({"type": "task/fetch", "data": card_type.id}),
            )
```

I have no idea why... does it break for your guys if you make this change?

Reply:
No

Reply:
Sorry.  Never tried that.  Shouldn't be necessary.  We can't explain why of the 3 of us, you have problems without it.

Reply:
Seems to work with it though.  Change pushed.